container/archive.go

```diff
@@ -6,13 +6,17 @@ import (
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/pkg/archive"
 	"github.com/docker/docker/pkg/system"
+	"github.com/pkg/errors"
 )
 
 // ResolvePath resolves the given path in the container to a resource on the
 // host. Returns a resolved path (absolute path to the resource on the host),
 // the absolute path to the resource relative to the container's rootfs, and
 // an error if the path points to outside the container's rootfs.
 func (container *Container) ResolvePath(path string) (resolvedPath, absPath string, err error) {
+	if container.BaseFS == nil {
+		return "", "", errors.New("ResolvePath: BaseFS of container " + container.ID + " is unexpectedly nil")
```

Comment:
Wonder if we should prefix all of these with the function-name

Reply:
The chance of such error is quite low (i.e. it should not happen with the current code base, as far as I was able to see), so normally a user won't even see it. If they do see it, though, it is kind of an internal error (as in "this software is buggy"), and having some context would be great to ease the debugging. Ideally, I would even add a stack trace here...